
#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################
mode: "neural" # classic  neural
model_path: "/home/nv/px4_policy_deploy/src/px4ctrl/src/models/actor_epoches_540.pt" # TODO: Set the path of the model file.

mass        : 0.752 # kg  0.7 0.882 0.6 0.7774 0.8865
gra         : 9.81
pose_solver : 0     # 0:From ZhepeiWang (drag & less singular) 1:From ZhepeiWang, 2:From rotor-drag  
policy_modality : 0 # 0:state-based (using odom), from wdww 1:pixel-based (using image) 
ctrl_freq_max   : 60.0  # 100  60
use_bodyrate_ctrl: true
max_manual_vel: 1.0
max_angle: 80  # Attitude angle limit in degree. A negative value means no limit.
low_voltage: 13.2 # 4S battery 6.6

#kdkd to cheak time syschron
# drone_id: 2 # 
# formation_num: 4 # 
#kdkd collition check
collision_threshold: 2500  # line_acc > 10 ,emergency kill
#kdkd takoff time limitation, normail:(t_lower_bound,t_upper_bound) 
t_lower_bound: 0.0  
t_upper_bound: 16.0 
#kdkd the tolerate offset between vins_odom and hover_position
hover_offset_x : 200.0 
hover_offset_y : 200.0 
hover_offset_z : 200.0

#kdk the tolerate offset of position when take_off, z:( 0,height + takeoff_offeset_z) 
takeoff_offeset_z_h : 1.0
takeoff_offeset_z_l : 0.7

# # the same as the max_vel of planner
# drone_max_vel : 1.0

# the tolerance of current pos and CMD_POS  
ctrl_tolerance_drift : 1.0
# maximum VINS_VEL during takeoff
takeoff_hover_max_vins_vel : 4
#
optical_avaliable_h : 2.0

rc_reverse: # *
    roll: false
    pitch: false
    yaw: false
    throttle: true

auto_takeoff_land:
    enable: true
    enable_auto_arm: true
    no_RC: false
    takeoff_height: 1.0 # m
    takeoff_land_speed: 0.25 # m/s 0.3

thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    print_value: false # display the value of “thr_scale_compensate” or “hover_percentage” during thrust model estimating.
    accurate_thrust_model: true  # This can always enabled if don't require accurate control performance :-)
    # accurate thrust mapping parameters
    K1: 0.6649 # Needs precise calibration!
    K2: 1.7979 # Needs precise calibration!
    K3: 0.6685 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    hover_percentage: 0.2  # Thrust percentage in Stabilize/Arco mode # *
    
    # new 2024.12.16 add light weight 700g 650g
    # K1: 0.6649 # Needs precise calibration!
    # K2: 1.7979 # Needs precise calibration!
    # K3: 0.6685 # Needs precise calibration!

    # new 2024.12.1
    # K1: 0.5920 # Needs precise calibration!
    # K2: 1.8500 # Needs precise calibration!
    # K3: 0.6856 # Needs precise calibration!

    # new 2024.11.1
    # K1: 0.5914 # Needs precise calibration!
    # K2: 1.8506 # Needs precise calibration!
    # K3: 0.6859 
    # # old
    # K1: 0.7583 # Needs precise calibration!
    # K2: 1.6942 # Needs precise calibration!
    # K3: 0.6786 # Needs precise calibration!

gain: 
    # Cascade PID controller. Recommend to read the code.
    Kp0: 4.5    # 4.0 5.0 5.0 
    Kp1: 4.5    # 4.0 5.0 5.0
    Kp2: 4.0    # 4.0 5.0 5.0
    Kv0: 6.0    # 4.0 5.0 6.0
    Kv1: 6.0    # 4.0 5.0 6.0
    Kv2: 7.0    # 5.0 6.0 7.0
    # ↓↓↓ No use now --
    Kvi0: 0.0
    Kvi1: 0.0
    Kvi2: 0.0
    Kvd0: 0.0
    Kvd1: 0.0
    Kvd2: 0.0
    # ↓↓↓ Only used in rate control mode.
    KAngR: 25.0
    KAngP: 24.0
    KAngY: 20.0

    # Kp0: 3.5
    # Kp1: 3.5 
    # Kp2: 3.5
    # Kv0: 2.5
    # Kv1: 2.5
    # Kv2: 2.5
    # # ↓↓↓ No use now --
    # Kvi0: 0.0
    # Kvi1: 0.0
    # Kvi2: 0.0
    # Kvd0: 0.0
    # Kvd1: 0.0
    # Kvd2: 0.0
    # # ↓↓↓ Only used in rate control mode.
    # KAngR: 20.0
    # KAngP: 20.0
    # KAngY: 20.0

rotor_drag:  
    x: 0.0  # The reduced acceleration on each axis caused by rotor drag. Unit:(m*s^-2)/(m*s^-1).
    y: 0.0  # Same as above
    z: 0.0  # Same as above
    k_thrust_horz: 0.0 # Set to 0 recommended... --

msg_timeout:
    odom: 0.2
    rc:   0.5
    cmd:  0.5
    imu:  0.5
    bat:  0.5
    optical_flow: 0.5
